src/main/java/de/tum/in/www1/artemis/service/iris/session/IrisHestiaSessionService.java

```diff
@@ -107,18 +108,14 @@ public CodeHint generateDescription(CodeHint codeHint) {
         }
     }
 
-    private IrisMessage generateSystemMessage() {
-        var irisMessage = new IrisMessage();
+    private IrisMessage generateSystemMessage(IrisSession session) {
+        var irisMessage = new IrisMessage(session);
```

Comment:
This is not needed. `irisMessageService.saveMessage`, already takes care of associating this properly. Setting it here is useless.